broker/src/main/java/org/apache/rocketmq/broker/topic/TopicRouteInfoManager.java

```diff
@@ -0,0 +1,276 @@
+import org.apache.rocketmq.remoting.exception.RemotingException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class TopicRouteInfoManager {
+
+    private static final long SEND_TIMEOUT = 3000L;
+    private static final long LOCK_TIMEOUT_MILLIS = 3000L;
+    private static final InternalLogger log = InternalLoggerFactory.getLogger(LoggerName.BROKER_LOGGER_NAME);
+
+    private final Lock lockNamesrv = new ReentrantLock();
+    private final ConcurrentMap<String/* Topic */, TopicRouteData> topicRouteTable = new ConcurrentHashMap<>();
+    private final ConcurrentMap<String/* Broker Name */, HashMap<Long/* brokerId */, String/* address */>> brokerAddrTable =
+        new ConcurrentHashMap<>();
+    private final ConcurrentMap<String/* topic */, TopicPublishInfo> topicPublishInfoTable = new ConcurrentHashMap<>();
+
+    private final ConcurrentHashMap<String, Set<MessageQueue>> topicSubscribeInfoTable = new ConcurrentHashMap<>();
+
+    private ScheduledExecutorService scheduledExecutorService;
+    private BrokerController brokerController;
+
+    public TopicRouteInfoManager(BrokerController brokerController) {
+        this.brokerController = brokerController;
+    }
+
+    public void start() {
+        this.scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+            @Override
+            public Thread newThread(Runnable r) {
+                return new Thread(r, "TopicRouteInfoManagerScheduledThread");
+            }
+        });
+
+        this.scheduledExecutorService.scheduleAtFixedRate(() -> {
+            try {
+                updateTopicRouteInfoFromNameServer();
+            } catch (Exception e) {
+                log.error("ScheduledTask: failed to pull TopicRouteData from NameServer", e);
+            }
+        }, 1000, this.brokerController.getBrokerConfig().getLoadBalancePollNameServerInterval(), TimeUnit.MILLISECONDS);
+    }
+
+    private void updateTopicRouteInfoFromNameServer() {
+        final Set<String> topicSetForAssignmentMgr = this.brokerController.getAssignmentManager().getTopicSetForAssignment();
+        final Set<String> topicSetForEscapeBridge = this.topicRouteTable.keySet();
+        final Set<String> topicsAll = Sets.union(topicSetForAssignmentMgr, topicSetForEscapeBridge);
+
+        for (String topic : topicsAll) {
+            boolean isNeedUpdatePublishInfo = topicSetForEscapeBridge.contains(topic);
+            boolean isNeedUpdateSubscribeInfo = topicSetForAssignmentMgr.contains(topic);
```

Comment:
isNeedUpdatePublishInfo and isNeedUpdateSubscribeInfo must be true here, so it is not necessary to judge.